packages/iov-bns/src/types.ts

```diff
@@ -23,6 +23,89 @@ export interface CashConfiguration {
   readonly minimalFee: Amount;
 }
 
+// Governance
+
+export interface Elector {
+  readonly address: Address;
+  /** The voting weight of this elector. Max value is 65535 (2^16-1). */
+  readonly weight: number;
+}
+
+export interface Electorate {
+  readonly id: number;
+  readonly version: number;
+  readonly admin: Address;
+  readonly title: string;
+  /** An unordered map from address to elector */
+  readonly electors: { readonly [index: string]: Elector };
```

Comment:
Right. But since `Electorate` data only flows from blockchain -> client (with no way to create a new `Electorate` from the client), we only get valid data in here (via decodeElectorate).

Expressing `electors` as a list as it is done in the weave codec indicates that it has a fixed order and addresses can occur multiple times, so this duplication is probably the smaller issue.

Reply:
One option would be to make this a map from address to weight.

Reply:
Yes, since we don't need to mirror the `Elector` type from weave (which I did), we can just use

```ts
interface ElectorProperties {
   readonly weight: number;
}
type Electors = { readonly [index: string]: ElectorProperties }
```